### PR TITLE
Override `equals` in ApplicantData

### PR DESCRIPTION
**Pull request by natsid**
Helpful for comparing ApplicantData objects in tests.

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Works toward #100

### CODE REVIEW

File: universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java [hunk 1]
```diff
@@ -62,4 +64,18 @@ public void setCreatedTime(Instant i) {
   public String asJsonString() {
     return this.jsonData.jsonString();
   }
+
+  @Override
+  public boolean equals(@Nullable Object object) {
+    if (object instanceof ApplicantData) {
+      ApplicantData that = (ApplicantData) object;
+      return Objects.equals(this.jsonData.jsonString(), that.jsonData.jsonString());
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java:72]: suggestion (blocking): Add a comment explaining why we can't use DocumentContext here

File: universal-application-tool-0.0.1/build.sbt [hunk 2]
```diff
@@ -34,6 +34,9 @@ lazy val root = (project in file("."))
       "org.awaitility" % "awaitility" % "4.0.1" % Test,
       "org.mockito" % "mockito-core" % "3.1.0" % Test,
 
+      // More testing
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/build.sbt]:
```suggestion
      // EqualsTester
```

File: universal-application-tool-0.0.1/build.sbt [hunk 3]
```diff
@@ -34,6 +34,9 @@ lazy val root = (project in file("."))
       "org.awaitility" % "awaitility" % "4.0.1" % Test,
       "org.mockito" % "mockito-core" % "3.1.0" % Test,
 
+      // General testing libraries
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/build.sbt:37]: suggestion (non-blocking): Consider linking to the testing library documentation https://javadoc.io/doc/com.google.guava/guava-testlib/latest/index.html

**Review by carolinedanzi — APPROVED** [review]
Nice work, thank you for getting EqualsTester working! This looks great

File: universal-application-tool-0.0.1/build.sbt (hunk 3 above)
Comment (natsid) [reply universal-application-tool-0.0.1/build.sbt:37]: Good idea! Done

File: universal-application-tool-0.0.1/build.sbt (hunk 2 above)
Comment (natsid) [reply universal-application-tool-0.0.1/build.sbt]: Done

File: universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java (hunk 1 above)
Comment (natsid) [reply universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java:74]: Done

**Pull request merged**